lib/computer_player.rb

```diff
@@ -1,10 +1,84 @@
 class ComputerPlayer
   attr_reader :mark
-  def initialize(mark:)
+  def initialize(mark:, difficulty:)
     @mark = mark
+    @difficulty = difficulty
```

Comment:
I like the idea of passing the difficulty here. I think you could take it one step further and encapsulate some of this "[strategy](https://refactoring.guru/design-patterns/strategy/ruby/example)" in it's own class. That way player doesn't need to know about the details of what the strategy is, just that it has one, and there is a method on it that it can call to get a move.